tools/src/main/scala/scala/scalanative/codegen/llvm/compat/os/UnixCompat.scala

```diff
@@ -54,41 +49,21 @@ private[codegen] class UnixCompat(codegen: AbstractCodeGen)
     indent()
     line(s"$rec = $landingpad")
     line(s"$r0 = extractvalue $excRecTy $rec, 0")
-    line(s"$r1 = extractvalue $excRecTy $rec, 1")
-    line(s"$id = $typeid")
-    line(s"$cmp = icmp eq i32 $r1, $id")
-    line(s"br i1 $cmp, label %$excsucc, label %$excfail")
```

Comment:
We don't do any type check. We assume any exception is a java.lang.Throwable and we handle it.